### ISSUE TITLE
Can't compile Godot3 on Linux (Javascript Export template)

### ISSUE THREAD

**Issue by Alexia-AT-Digitecnology** [godotengine/godot#14670]
It gives these error:

```
[ 75%] em++ -o scene/3d/room_instance.javascript.opt.bc -c -O3 -w -DNDEBUG -DZSTD_STATIC_LINKING_ONLY -DFT2_BUILD_LIBRARY -DFREETYPE_ENABLED -DRECAST_ENABLED -DSVG_ENABLED -DHAVE_MNTENT -DJAVASCRIPT_ENABLED -DUNIX_ENABLED -DPTHREAD_NO_RENAME -DTYPED_METHOD_BIND -DNO_THREADS -DGLES3_ENABLED -fno-exceptions -DNO_SAFE_CAST -fno-rtti -DJAVASCRIPT_EVAL_ENABLED -DPTRCALL_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -DXML_ENABLED -Icore -Icore/math -Ieditor -Idrivers -I. -Iplatform/javascript -Ithirdparty/zstd -Ithirdparty/zstd/common -Ithirdparty/zlib -Ithirdparty/freetype -Ithirdparty/freetype/include -Ithirdparty/libpng -Ithirdparty/recastnavigation/Recast -Ithirdparty/recastnavigation/Recast/Include -Ithirdparty/nanosvg scene/3d/room_instance.cpp
scene/3d/voxel_light_baker.cpp:1032:15: error: call to 'pow' is ambiguous
                float att = Math::pow(1.0 - angle / p_spot_angle, p_spot_attenuation);
                            ^~~~~~~~~
core/math/math_funcs.h:99:32: note: candidate function
        static _ALWAYS_INLINE_ double pow(double p_x, double p_y) { return ::pow(p_x, p_y); }
                                      ^
core/math/math_funcs.h:100:31: note: candidate function
        static _ALWAYS_INLINE_ float pow(float p_x, float p_y) { return ::powf(p_x, p_y); }
                                     ^
scene/3d/voxel_light_baker.cpp:1641:28: error: use of undeclared identifier 'rand'; did you mean 'Math::rand'?
                float random_angle1 = (((rand() % 65535) / 65535.0) * 2.0 - 1.0) * spread;
                                         ^~~~
                                         Math::rand
core/math/math_funcs.h:225:18: note: 'Math::rand' declared here
        static uint32_t rand();
                        ^
scene/3d/voxel_light_baker.cpp:1643:27: error: use of undeclared identifier 'rand'; did you mean 'Math::rand'?
                float random_angle2 = ((rand() % 65535) / 65535.0) * Math_PI * 2.0;
                                        ^~~~
                                        Math::rand
core/math/math_funcs.h:225:18: note: 'Math::rand' declared here
        static uint32_t rand();
                        ^
3 errors generated.
ERROR:root:compiler frontend failed to generate LLVM bitcode, halting
scons: *** [scene/3d/voxel_light_baker.javascript.opt.bc] Error 1
```

Compiled on Digitecnology Server for bringing Godot 3 builds. Commit id: f3ad142

**Comment by AlmightyScientist** [comment]
We're aware of this issue :p 
GI integration seems troublesome.

**Comment by akien-mga** [comment]
That's fixed already by #14657.
